compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -564,6 +582,20 @@ isFusableWithConsumer(OpOperand &fusedOperand,
     return false;
   }
 
+  // Insert slice ops should always be fused with their producers.
+  if (auto insertSliceOp = dyn_cast<tensor::InsertSliceOp>(consumer)) {
+    // TODO: Enable multi-use slice source fusion.
+    Value source = insertSliceOp.getSource();
+    if (!source.hasOneUse() || source.getDefiningOp() != producer) {
+      return false;
+    }
+    // Fuse in `insert_slice` consumer operations if destination is a fill.
+    // TODO: This can be generalized, but destination cannot be a
+    // `arith.constant` or other constant-like objects. `linalg.fill` captures a
+    // common case of pad generalization.
+    return insertSliceOp.getDest().getDefiningOp<linalg::FillOp>();
```

Comment:
What goes wrong for `tensor.empty`? That is the other common case I would expect.

Reply:
inserting into an empty is kind of strange. But we can generalize as needed (as the comment says)

Reply:
it happens very often for concatenations (same with insert_slice -> insert_slice). Anyway doesn't have to change in this PR.

Reply:
yeah, maybe... I tend to slow walk such decisions... if you find a use case for `empty`, I can add it (will try anyway after I land this PR).